### PR TITLE
Microsoft.HealthcareApis EventGrid Spec

### PR DESCRIPTION
**Pull request by kannanMSFT**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [X] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
        <br/>**Mar 2022**

  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
   <br/>**Feb 2022**

  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [X] No refresh required for updates in this PR

### Contribution checklist:
- [X] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [X] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [X] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,8 +1,2143 @@
 {
   "name": "azure-rest-api-specs-tests",
   "version": "0.1.0",
-  "lockfileVersion": 1,
+  "lockfileVersion": 2,
   "requires": true,
+  "packages": {
```
Comment (JoshLove-msft) [inline package-lock.json:6]: Why is this file being changed?

Comment (kannanMSFT) [reply package-lock.json:6]: Looks like running prettier validation locally created this. Let me revert this one.

Comment (kannanMSFT) [reply package-lock.json]: Done.

File: specification/eventgrid/data-plane/Microsoft.HealthcareApis/stable/2018-01-01/HealthcareApis.json [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        },
+        "resourceFhirId": {
+          "description": "Id of HL7 FHIR resource.",
+          "type": "string"
+        },
+        "resourceVersionId": {
+          "description": "VersionId of HL7 FHIR resource. It changes when the resource is created, updated, or deleted.",
+          "type": "integer",
+          "format": "int64"
+        }
```
Comment (ms-teli) [inline specification/eventgrid/data-plane/Microsoft.HealthcareApis/stable/2018-01-01/HealthcareApis.json:30]: As for now, we should have these 4 as critical properties of the event payload. They represent exact same states. Can we use a $ref tag to reduce duplication and make it easier for maintenance?

Example: 
- https://github.com/Azure/azure-rest-api-specs/blob/679887ace44697c726aba8d2814ee415a5d25e6f/specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json#L663

Ref: 
- https://swagger.io/docs/specification/using-ref/
- https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/

File: specification/eventgrid/data-plane/Microsoft.HealthcareApis/stable/2018-01-01/examples/fhir_resource_created.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
+{
+  "id": "e4c7f556-d72c-e7f7-1069-1e82ac76ab41",
+  "topic": "/subscriptions/4805176c-73e4-4b8a-88cf-ea7d61b176a7/resourceGroups/rg-ga-ui-wsstatus-central/providers/Microsoft.HealthcareApis/workspaces/wsga",
```
Comment (ms-teli) [inline specification/eventgrid/data-plane/Microsoft.HealthcareApis/stable/2018-01-01/examples/fhir_resource_created.json:3]: let's hide the details for those ids and names. for example:

/subscriptions/{subscription-id}/resourceGroups/{resource-group-name}/providers/Microsoft.HealthcareApis/workspaces/{workspace-name}

File: specification/eventgrid/data-plane/Microsoft.HealthcareApis/stable/2018-01-01/examples/fhir_resource_created.json [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  "subject": "wsga-fs-ga-wsstatus-r4-01.fhir.azurehealthcareapis.com/Patient/e0a1f743-1a70-451f-830e-e96477163902",
+  "data": {
+    "resourceType": "Patient",
+    "resourceFhirAccount": "wsga-fs-ga-wsstatus-r4-01.fhir.azurehealthcareapis.com",
```
Comment (ms-teli) [inline specification/eventgrid/data-plane/Microsoft.HealthcareApis/stable/2018-01-01/examples/fhir_resource_created.json:7]: same, let's use mocking strings for the FHIR account endpoint, e.g: {your-fhir-account}.fhir.azurehealthcareapis.com

File: specification/eventgrid/data-plane/Microsoft.HealthcareApis/stable/2018-01-01/HealthcareApis.json [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        "resourceFhirAccount": {
+          "description": "Domain name of FHIR account for this resource.",
+          "type": "string"
+        },
+        "resourceFhirId": {
+          "description": "Id of HL7 FHIR resource.",
+          "type": "string"
+        },
+        "resourceVersionId": {
+          "description": "VersionId of HL7 FHIR resource. It changes when the resource is created, updated, or deleted.",
```
Comment (ms-teli) [inline specification/eventgrid/data-plane/Microsoft.HealthcareApis/stable/2018-01-01/HealthcareApis.json:27]: Strictly speaking, it should be 
"VersionId of HL7 FHIR resource. It changes when the resource is created, updated, or deleted (soft-deletion)."

**Review by ms-teli — CHANGES_REQUESTED** [review]
Looks good overall, put a few minor comments.

**Review by ms-teli — APPROVED** [review]
Looks good to me.

**Review by lmazuel — APPROVED** [review]

**Review by rakshith91 — APPROVED** [review]

**Pull request merged**